src/services/ConfigService.jsx

```diff
@@ -0,0 +1,20 @@
+export class ConfigService {
+  constructor({ apiClient }) {
+    this.apiClient = apiClient
+  }
+
+  getEndpoint({ siteName, isHomepage, isSettings }) {
+    if (isHomepage) return `/sites/${siteName}/homepage`
+    if (isSettings) return `/sites/${siteName}/settings`
+  }
```

Comment:
i think this is p prone to error because callers have to explicitly pass in a `isSettings` or `isHomepage`, failing which this returns undefined. 

Something that we might wanna consider here is who decides the final API endpoint - if it's the caller of this function (indirectly through the hook), we can simplify and just pass in an endpoint directly like so: 

```js
getEndpoint({siteName, endPoint}) {
    return `/sites/${siteName}/${endpoint}`
}
```

this makes it clearer and also hands control back to the caller of the hook, which seems to be what is happening already.

Reply:
This makes sense in this context, and I'm open to changing it here, but for our other endpoints e.g. [page services](https://github.com/isomerpages/isomercms-frontend/blob/develop/src/services/PageService.jsx), the parameterisation of the endpoint is passed directly from the url parameters. 

For example, for page: https://staging-cms.isomer.gov.sg/sites/a-test-v4/folders/default/editPage/Lorem%20Ipsum.md, the url parameters will be as follows:
```
{
  siteName: 'a-test-v4',
  collectionName: 'default',
  fileName: 'Lorem%20Ipsum.md',
}
```

And this url parameterisation directly populates the `getEndpoint(params)` computation. By matching the url parameters with the endpoint parameterised callbacks, we can simplify most of endpoint calling on the frontend as there's no messy logic on computation what the endpoint is on each hook caller. In such cases, I would argue against passing back the control over the endpoint to the caller of the hook as it would create more messy logic in the caller. 

I wonder if the more elegant solution is to put something in the url parameter to parameterise this endpoint?

Reply:
What do you think!!

Reply:
sflr!! wouldn't passing the config object amount to returning control back to the caller also? when we call with the url params, it's the caller that decides what gets pinged isn't it? 

fwiw the reason i'm hesitant to put the endpoint into the url params is that the parameter always fixed at this page, whereas url params might change (and break downstream components that rely on it) but i do agree that it will be alot cleaner!

Reply:
fixed in d72b64, d196d3f, 220b6ca